addons/stock/models/stock_move.py

```diff
@@ -503,7 +503,7 @@ def _set_lot_ids(self):
                     move_line_vals['product_uom_id'] = move.product_id.uom_id.id
                     move_line_vals['qty_done'] = 1
                     move_lines_commands.append((0, 0, move_line_vals))
-                else:
+                elif lot.product_id.tracking == "serial":
```

Comment:
You could do it at the start of the function no? So it avoid useless computation
```python3
if move.product_id..tracking != 'serial':
    continue
```